### PR TITLE
update hook Typescript types

### PR DESCRIPTION
**Pull request by ricfang**

### CODE REVIEW

File: packages/redux-query-react/index.d.ts [hunk 1]
```diff
@@ -45,20 +48,17 @@ declare module 'redux-query-react' {
   }
   export type ReduxQueryProvider = React.ComponentType<ProviderProps>;
 
-  export type ActionPromise<TEntities = Entities> =
-    | Promise<ActionPromiseValue<TEntities>>
-    | undefined;
   type RunMutation<TEntities = Entities, TArgs = never> = (
     ...args: TArgs
-  ) => ActionPromise<TEntities>;
+  ) => Promise<ActionPromiseValue<TEntities>>;
```
Comment (ricfang) [inline packages/redux-query-react/index.d.ts:53]: should match: https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-mutation.js#L13-L15

File: packages/redux-query-react/index.d.ts [hunk 2]
```diff
@@ -22,7 +22,10 @@ declare module 'redux-query-react' {
   }
   export type ForceRequestCallback<TEntities = Entities> = () => Promise<
     ActionPromiseValue<TEntities>
-  >;
+  > | void;
```
Comment (ricfang) [inline packages/redux-query-react/index.d.ts:25]: https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-request.js#L15-L17

File: packages/redux-query-react/index.d.ts [hunk 3]
```diff
@@ -22,7 +22,10 @@ declare module 'redux-query-react' {
...
+
+  export type ForceRequestsCallback = () => void;
```
Comment (ricfang) [inline packages/redux-query-react/index.d.ts:27]: technically its typed the same as `useRequest`

https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-requests.js#L49-L51

but `forceRequest` doesn't return anything:

https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-requests.js#L111-L118

**Review by rctbusk — APPROVED** [review]
LGTM

**Review by kelvin-lu — APPROVED** [review]

**Comment by ricfang** [comment]
hey @siegcollado I noticed you changed

```
export type RunMutation<TEntities> = <TEntities>(
    ...args: any	
  ) => Promise<ActionPromiseValue<TEntities>>;
```

to

```
 export type ActionPromise<TEntities = Entities> =
    | Promise<ActionPromiseValue<TEntities>>
    | undefined;
  type RunMutation<TEntities = Entities, TArgs = never> = (
    ...args: TArgs
  ) => ActionPromise<TEntities>;
```

based on https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-mutation.js#L13-L15, I think the return type should be what was there before (never returning `undefined`)

do you have any more context on that change?

**Review by ManThursday — APPROVED** [review]
lgtm

@rctbusk I think this should be a major version bump due to the removal of `ActionPromise`

**Comment by rctbusk** [comment]
@ManThursday I agree this is not backwards compatible. @ricfang can we add back the ActionPromise for now? Just have the export so that we don't break. Will make the upgrade a minor version release. I will remove the type in the next major release. (which hopefully will be soon).

**Comment by ricfang** [comment]
@rctbusk added back the export and removed `undefined`

**Pull request merged**